api/src/main/java/io/druid/timeline/DataSegmentUtils.java

```diff
@@ -0,0 +1,188 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.timeline;
+
+import com.google.common.base.Function;
+import com.metamx.common.logger.Logger;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.util.Objects;
+
+/**
+ * identifier to DataSegment.
+ */
+public class DataSegmentUtils
+{
+  private static final Logger LOGGER = new Logger(DataSegmentUtils.class);
+
+  public static Function<String, Interval> INTERVAL_EXTRACTOR(final String datasource)
+  {
+    return new Function<String, Interval>()
+    {
+      @Override
+      public Interval apply(String identifier)
+      {
+        return valueOf(datasource, identifier).getInterval();
+      }
+    };
+  }
+
+  // ignores shard spec
+  public static SegmentElements valueOf(String dataSource, String identifier)
+  {
+    SegmentElements segmentDesc = parse(dataSource, identifier);
+    if (segmentDesc == null) {
+      throw new IllegalArgumentException("Invalid identifier " + identifier);
+    }
+    return segmentDesc;
+  }
+
+  private static SegmentElements parse(String dataSource, String identifier)
+  {
+    if (!identifier.startsWith(dataSource + DataSegment.delimiter)) {
+      return null;
+    }
+    String remaining = identifier.substring(dataSource.length() + 1);
+    String[] splits = remaining.split(DataSegment.delimiter);
+    if (splits.length < 2) {
+      return null;
+    }
+
+    DateTimeFormatter formatter = ISODateTimeFormat.dateTime();
+    DateTime start = formatter.parseDateTime(splits[0]);
+    DateTime end = formatter.parseDateTime(splits[1]);
+    String version = splits.length > 2 ? splits[2] : null;
+    String trail = splits.length > 3 ? join(splits, DataSegment.delimiter, 3, splits.length) : null;
+
+    return new SegmentElements(
+        dataSource,
+        new Interval(start.getMillis(), end.getMillis()),
+        version,
+        trail
+    );
+  }
+
+  public static String withInterval(final String dataSource, final String identifier, Interval newInterval)
+  {
+    SegmentElements segmentDesc = DataSegmentUtils.parse(dataSource, identifier);
+    if (segmentDesc == null) {
+      // happens for test segments which has invalid segment id.. ignore for now
+      LOGGER.warn("Invalid segment identifier " + identifier);
+      return identifier;
+    }
+    return segmentDesc.withInterval(newInterval).toString();
+  }
+
+  static class SegmentElements
+  {
+    private final String dataSource;
+    private final Interval interval;
+    private final String version;
+    private final String trail;
+
+    public SegmentElements(String dataSource, Interval interval, String version, String trail)
+    {
+      this.dataSource = dataSource;
+      this.interval = interval;
+      this.version = version;
+      this.trail = trail;
```

Comment:
Yes, currently. I think there would/could be more informations appended in identifier later.